contracts/crashpunks-v2-friedger.clar

```diff
@@ -0,0 +1,345 @@
+;; Interface definitions
+(impl-trait .nft-trait.nft-trait)
+(impl-trait .operable.operable)
+;; (impl-trait .transferable.transferable)
+
+;; contract variables
+
+(define-data-var administrator principal 'SP3N4AJFZZYC4BK99H53XP8KDGXFGQ2PRSQP2HGT6)
+
+
+;; TODO: MAKE SURE THIS MINT COUNTER IS CORRECT. SHOULD BE THE MINT-COUNTER FROM V1. DOUBLE CHECK IF OFF BY 1 ERROR
+(define-data-var mint-counter uint u5721)
+
+;; addresses to receive mint fee
+(define-data-var collection-mint-addresses (list 4 principal) (list))
+;; percent mint fee each address receives
+(define-data-var collection-mint-shares (list 4 uint) (list))
+
+;; addresses to receive royalty fees
+;; first element at this list is ignored because the nft owner will be used
+(define-data-var collection-royalty-addresses (list 10 principal) (list))
+;; percent royalty fee each address receives
+(define-data-var collection-royalty-shares (list 10 uint) (list))
+
+;; TODO: update this
+(define-data-var base-uri (string-ascii 80) "ipfs://Qmad43sssgNbG9TpC6NfeiTi9X6f9vPYuzgW2S19BEi49m/{id}")
```

Comment:
base-uri suggests that some specific elements are added. However, this is already the full token-uri

Is the limit to 80 on purpose?

Reply:
not sure, i used from megapont. will ask